### PR TITLE
login field

### PR DESCRIPTION
**Pull request by paul1124**
renamed branch fe_login to feature/login

### CODE REVIEW

**Comment by clockwerkz** [comment]
Looks good, Paul :)

**Review by terrancexin — CHANGES_REQUESTED** [review]
let's use React Hooks for Login.jsx

File: client/src/Components/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
+// import React, { Component } from 'react'
+import React, { useState } from 'react';
+
+export default function Login() {
+  const [ username, setUsername ] = useState();
+  const [ password, setPassword ] = useState();
+
+  return (
+    <div className="login">
+      <LoginForm setUsername={setUsername} setPassword={setPassword} />
```
Comment (terrancexin) [inline client/src/Components/Login.jsx]: why are you passing in setState functions?

File: client/src/Components/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      {console.log(username, password)}
+    </div>
+  )
+}
+
+function LoginForm({ setUsername, setPassword }) {
```
Comment (terrancexin) [inline client/src/Components/Login.jsx]: you should merge these into one component, the setPassword and setPw are reusable.

**Comment by paul1124** [comment]
> why do you need both setPw and setPassword?

Since I need to submit both username and password when I'm trying to bypass the login page. The setPw works just like value, where I'm trying to only handleChange for the password input. I can't do use [value, setValue] for both username and password, because handleSubmit is going to take account of both username and password, and setValue is going to change username only via handleChange.

**Comment by terrancexin** [comment]
```js
import React, { useState } from 'react';

export default function Login() {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  return (
    <div className="login">
      <form onSubmit={handleSubmit}>
        <label>Username: </label>
        <input
          type="text"
          value={username}
          onChange={e => setUsername(e.target.value)}
        />
        <label>Password: </label>
        <input
          type="password"
          value={password}
          onChange={e => setPassword(e.target.value)}
        />
        <button onClick={handleSubmit}>Log In</button>
      </form>
    </div>
  )

  function handleSubmit(e) {
    e.preventDefault();
  
   // Good job with UI validation before we're sending back the data to the UI, we should connect further to chat about different approaches for error validations. It's very interesting to think about.
    if (username.length < 4 && password.length < 4) return;

  // and in this submit function you can make a HTTP POST request to send the user data back to our backend server
    axios.post('/auth', { username, password })
      .then(data => console.log(data))
  }
}
```

let me know if you have any questions on the code or comment.

**Review by nroode — COMMENTED** [review]
looks good Paul - though the imports weren't working for me since they referenced the 'Components' folder (upper case) instead of 'components'

**Review by clockwerkz — CHANGES_REQUESTED** [review]
Looking good, Paul!
Yeah the only thing is you have "Components" vs "components" I think we had "components" already, but let's choose one folder and use that moving forward. :)

**Comment by terrancexin** [comment]
👍

**Review by terrancexin — APPROVED** [review]

**Pull request merged**